5_data_munge/src/munge_data_files.R

```diff
@@ -0,0 +1,158 @@
+f_to_c <- function(f) {
+  c <- (f - 32) * (5/9)
+  return(c)
+}
+
+munge_wqp_withdepths <- function(in_ind, min_value, max_value, max_daily_range, out_ind) {
+  dat <- readRDS(sc_retrieve(in_ind)) %>%
+    filter(!is.na(`ResultMeasure/MeasureUnitCode`)) %>%
+    filter(ActivityMediaName %in% 'Water') %>%
+    filter(!is.na(`ActivityDepthHeightMeasure/MeasureValue`)|
+                         !is.na(`ResultDepthHeightMeasure/MeasureValue`)|
+                         !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`)|
+                         !is.na(`ActivityBottomDepthHeightMeasure/MeasureValue`)) %>% 
+    filter(ActivityStartDate > as.Date('1800-01-01') & ActivityStartDate < as.Date('2020-01-01'))
+  
+  # create depth column
+  dat <- mutate(dat, sample_depth = case_when(
+    !is.na(`ActivityDepthHeightMeasure/MeasureValue`) ~ `ActivityDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & !is.na(`ResultDepthHeightMeasure/MeasureValue`) ~ `ResultDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityBottomDepthHeightMeasure/MeasureValue`
+  ))
+  
+  dat <- mutate(dat, sample_depth_unit_code = case_when(
+    !is.na(`ActivityDepthHeightMeasure/MeasureValue`) ~ `ActivityDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & !is.na(`ResultDepthHeightMeasure/MeasureValue`) ~ `ResultDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityBottomDepthHeightMeasure/MeasureUnitCode`
+  ))
+  
+
+  # now reduce to daily measures
+  dat_reduced <- dat %>%
+    filter(ResultMeasureValue > min_value) %>%
+    filter(grepl('deg C|deg F', `ResultMeasure/MeasureUnitCode`, ignore.case = TRUE)) %>%
+    mutate(ResultMeasureValue = ifelse(grepl('deg F', `ResultMeasure/MeasureUnitCode`, ignore.case = TRUE),
+                                       f_to_c(ResultMeasureValue), ResultMeasureValue)) %>%
+    mutate(`ResultMeasure/MeasureUnitCode` = 'deg C') %>%
+    filter(ResultMeasureValue < max_value)
+    
+  dat_daily <- group_by(dat_reduced, MonitoringLocationIdentifier, ActivityStartDate, sample_depth) %>%
```

Comment:
Hmm, I hadn't realized that WQP provided depth-specific values for water quality. Does this happen in rivers as well as lakes/reservoirs? If so, seems we'll want to accommodate this when collecting observations for stream temperature modeling.

Reply:
Yeah for now, without much digging into metadata, I've just handled depths and no depths records independently. My guess is that it's mostly lakes/reservoirs, with streams mixed in. The final pushed daily temperature product does not have these observations (obs with depth) included because I wasn't quite ready to tackle that problem.